android/java/org/chromium/chrome/browser/crypto_wallet/activities/NftDetailActivity.java

```diff
@@ -0,0 +1,212 @@
+    private TextView mNftDetailTitleView;
+    private TextView mNftNameView;
+    private ViewGroup mNftDescriptionLayout;
+    private Toolbar mToolbar;
+
+    private PortfolioModel.Erc721MetaData mErc721MetaData;
+
+    @Override
+    protected void triggerLayoutInflation() {
+        setContentView(R.layout.activity_nft_detail);
+
+        Intent intent = getIntent();
+        if (intent != null) {
+            mChainId = intent.getStringExtra(CHAIN_ID);
+            mNftName = intent.getStringExtra(ASSET_NAME);
+            mContractAddress = intent.getStringExtra(ASSET_CONTRACT_ADDRESS);
+            mNftTokenHex = intent.getStringExtra(NFT_TOKEN_ID_HEX);
+            mNftTokenId = Utils.hexToIntString(mNftTokenHex);
+            mErc721MetaData =
+                    (PortfolioModel.Erc721MetaData) intent.getSerializableExtra(NFT_META_DATA);
+        }
+
+        // Calculate half screen height and assign it to NFT image view,
+        // and "Image not found" text view (that will pop up in case of issues).
+        int halfScreenHeight = AndroidUtils.getScreenHeight() / 2;
+
+        mNftImageView = findViewById(R.id.nft_detail_image);
+        mNftImageView.getLayoutParams().height = halfScreenHeight;
+
+        mImageNotAvailableText = findViewById(R.id.nft_detail_image_not_available);
+        mImageNotAvailableText.getLayoutParams().height = halfScreenHeight;
+
+        mDescriptionContentView = findViewById(R.id.description_content);
+
+        mToolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(mToolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        mBtnSend = findViewById(R.id.btn_send);
+        // TODO(simone): Enable if it's the NFT owner.
+        mBtnSend.setVisibility(View.GONE);
+
+        mNftDetailTitleView = findViewById(R.id.nft_detail_title);
+        mNftDetailTitleView.setText(Utils.formatErc721TokenTitle(mNftName, mNftTokenId));
+
+        mNftNameView = findViewById(R.id.nft_name);
+        mNftNameView.setText(mNftName);
+
+        mNetworkNameView = findViewById(R.id.blockchain_content);
+
+        mTokenStandardView = findViewById(R.id.token_standard_content);
+        mTokenStandardView.setText(R.string.brave_wallet_nft_erc_721);
+
+        mTokenIdView = findViewById(R.id.token_id_content);
+
+        mNftDescriptionLayout = findViewById(R.id.nft_description);
+
+        setMetadata(mErc721MetaData);
+
+        BraveActivity braveActivity = BraveActivity.getBraveActivity();
+        assert braveActivity != null;
+
+        LiveDataUtil.observeOnce(
+                braveActivity.getWalletModel().getCryptoModel().getNetworkModel().mDefaultNetwork,
+                defaultNetwork -> {
+                    String networkName = defaultNetwork.chainName;
+                    mNetworkNameView.setText(networkName);
+                    String blockExplorerUrl = defaultNetwork.blockExplorerUrls.length > 0
+                            ? defaultNetwork.blockExplorerUrls[0]
```

Comment:
Non blocking: is there a reason this needs to be an array if we're just grabbing the first index here always?

Reply:
It's a `Mojom` generated `NetworkInfo` class to get response data of network as an object from backend so it cannot be changed as it's being used across platforms and since there can be 0,1 or 1+ values so array is the ideal type for this kind of data. Changing this would require changes across platforms, backend and there could be a use case where more than the first url is being used.

Reply:
That's a very good question!
The `blockExplorerUrls` is an array because some tokens may have multiple blockchain explorer endpoints, this could be useful in case one endpoint is down. Of course, we should add the required logic on top where we check if an endpoint is down and process the next one if present. (It could be an improvement to implement in the future).

Another constraint to consider is about our current configuration as well: `NetworkInfo.java` (`defaultNetwork`) is a Java class generated by the mojo binding generator. So the original type is a `struct` defined in `brave_wallet.mojom` [here](https://github.com/brave/brave-core/blob/a3503ed28bd810086cf930f355be0d34ba88770f/components/brave_wallet/common/brave_wallet.mojom#L834-L846). 
Not savvy enough to know how much room we have to modify these `mojom` files. I'm sure @SergeyZhukovsky could tell us more.